File: src/webapp/src/models/card.ts
import { merge } from 'lodash';
import { GenericError } from './exceptions';
import { Developer } from './developer';

export class Card {
  id = 0;
  name = '';
  description = '';
  owners: Array<Developer> = [];
  priority = 0;
  estimatedPoints = 0;
  state = 0;

  static create(data: any): Card {
    if (!data) throw new GenericError('data expected');

    const model = new Card();

    merge(model, data, {
      owners:
        data.owners != undefined
          ? data.owners.map((o: any) => Developer.create(o))
          : [],
    });

    return model;
  }
}

File: src/backend/Workboard.Domain/Model/Card.cs
namespace Workboard.Domain.Model;

public partial class Card
{
#nullable disable
    private Card()
    {
        _owners = new();
    }
#nullable enable

    private readonly List<Developer> _owners;

    public Card(string name, string description, int priority, decimal estimatedPoints, List<Developer> owners)
    {
        Name = name;
        Description = description;
        _owners = owners;
        Priority = priority;
        EstimatedPoints = estimatedPoints;
        State = CardState.NotStarted;
    }

    public int Id { get; private set; }
    public string Name { get; private set; }
    public string Description { get; private set; }
    public IEnumerable<Developer> Owners => _owners.AsReadOnly();
    public int Priority { get; private set; }
    public decimal EstimatedPoints { get; private set; }
    public int State { get; set; }
}
